ci-operator/step-registry/upi/conf/gcp/upi-conf-gcp-commands.sh

```diff
@@ -25,6 +25,18 @@ pushd ${dir}
 cp -t "${dir}" \
     "${SHARED_DIR}/install-config.yaml"
 
+### Read XPN config, if exists
+if [[ -s "${SHARED_DIR}/xpn.json" ]]; then
```

Comment:
Can we break these into their own steps? I think we want to avoid overloading these scripts & creating multiple control flows.

Reply:
Anything between `create manifests` and `create ignition-configs` generates too much data to try to store in SHARED_DIR... so it all needs to be in a single step. Therefore, only the step to make the cluster 'internal' would apply. IMHO it's not worth the complexity to try to break that out at this point in time.

Reply:
If we can't break into steps, another option to consider would be creating another parallel step which has some duplication. I think the simplicity benefits are worth considering.

Reply:
I disagree. We did this with templates, and changes made in one template didn't always make it into another... causing unnecessary drift. These variants are only a minor part of the much larger picture.

Reply:
I think I agree with @patrickdillon  on this one. 
The current step is different enough for example single dns zone for both public and private and other XPN changes  that it has made this step a lot more difficult to read and follow.
I think the CI and docs have different users in mind and trying to force them together isn't a good choice. 

But we can look at splitting them out as followup.